docs/concepts/Security-Best-Practices.md

```diff
@@ -0,0 +1,166 @@
+
+## Supply Chain Compromises
+
+The traditional definition of a supply chain comes from manufacturing, it is the chain of processes required to make and supply something. It includes planning, supply of materials, manufacturing, and retail. A software supply chain is similar, except instead of materials, it is code. Instead of manufacturing it is development. Instead of digging ore from the ground, code is sourced from suppliers, commercial or open source, and, in general, the open-source code comes from repositories. Adding code from a repository means your product takes a dependency on that code.
+
+One example of a software supply chain attack occurs when malicious code is purposefully added to a dependency, using the supply chain of that dependency to distribute the code to its victims. Supply chain attacks are real. There are many methods to attack a supply chain, from directly inserting malicious code as a new contributor, to taking over a contributor’s account without others noticing, or even compromising a signing key to distribute software that is not officially part of the dependency.
+
+Software supply chain attacks are in and of itself rarely the end goal, rather it is the beginning of an opportunity for an attacker to insert malware or provide a backdoor for future access.
+
+![The State of the Octoverse 2020 - Vulnerability Lifecycle](media/vulnerability-lifecycle.png)
+
+## Unpatched Software
+
+The use of open source today is significant and is not expected to slow down anytime soon. Given that we are not going to stop using open-source software, the threat to supply chain security is unpatched software. Knowing that, how can you address the risk that a dependency of your project has a vulnerability?
+
+- **Knowing what is in your environment.** This requires discovering your dependencies and any transitive dependencies to understand the risks of those dependencies such as vulnerabilities or licensing restrictions.
+- **Manage your dependencies.** When a new security vulnerability is discovered, you must determine whether you are impacted, and if so, update to the latest version and security patch available. This is especially important to review changes that introduce new dependencies or regularly auditing older dependencies.
+- **Monitor your supply chain.** This is by auditing the controls you have in place to manage your dependencies. This will help you enforce more restrictive conditions to be met for your dependencies.
+
+![The State of the Octoverse 2020 - Advisories](media/advisories.png)
+
+We will cover various tools & techniques that NuGet & GitHub provides that you can use today to address potential risks inside your project. 
+
+## Knowing what is in your environment
+
+### NuGet Dependency Graph
```

Comment:
Listing the packages in my graph doesn't provide information about their vulnerabilities.. what is the call for action here?  What should I do after I list them? IMHO this should be replaced by using tooling to indentify deprecated/vulnerable packages.

Reply:
I know this sounds very basic, but just knowing what dependencies you have & their versions is a best practice. For example knowing you don't use a package can help you remove it. Seeing a typo in a package might lead to a wrong dependency, etc.